packages/flame/lib/src/game/projector.dart

```diff
@@ -0,0 +1,88 @@
+import '../../extensions.dart';
+
+/// A simple interface to mark a class that can perform projection operations
+/// from one 2D euclidian coordinate space to another.
+///
+/// This can be a Viewport, a Camera or anything else that exposes such
+/// operations to the user.
+abstract class Projector {
+  /// Converts a vector in the screen space to the world space.
+  ///
+  /// This considers both the translation and scaling transformations.
+  Vector2 unprojectVector(Vector2 screenCoordinates);
+
+  /// Converts a vector in the world space to the screen space.
+  ///
+  /// This considers both the translation and scaling transformations.
+  Vector2 projectVector(Vector2 worldCoordinates);
+
+  /// Converts a vector representing a delta in the screen space to the world
+  /// space.
+  ///
+  /// This considers only the scaling transformation, as the translations are
+  /// cancelled in a delta transformation.
+  /// A delta can be a displacement (difference between two position
```

Comment:
With the docs the name delta is a little better, but I still feel that scaleVector makes more sense, wdyt?

Reply:
I think any of them works

Reply:
`scaleVector` seems good as well, I would normally prefer to focus on what it should be used for rather than what it does, to make users life easier.

Reply:
I think I still prefer `scaleVector`, this could be used for dimensions to for example, which isn't a delta. And `scaleVector` would work fine for both examples.

Reply:
Up to you!